include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,410 @@
+
+enum EulerRotationType
+{
+  Extrinsic = 0,
+  Intrinsic
+};
+
+const std::array<int, 3> DEFAULT_EULER_ANGLE_AXES = {0, 1, 2};
+const EulerRotationType DEFAULT_EULER_ANGLE_ROTATION_TYPE = Extrinsic;
+
+/**
+ * @brief Euler angle rotation object.
+ * @note Default convention is <b>X-Y-Z Extrinsic</b> Euler angles.
+ *
+ * @tparam Scalar[optional] numeric type
+ */
+template<typename Scalar = float>
+struct EulerAngles
+{
+  static_assert(std::is_arithmetic<Scalar>::value,
+                "Euler angles must have a numeric type.");
+
+  /**
+   * @brief Roll angle (default radians)
+   */
+  Scalar roll = 0;
+
+  /**
+   * @brief Pitch angle (default radians)
+   */
+  Scalar pitch = 0;
+
+  /**
+   * @brief Yaw angle (default radians)
+   */
+  Scalar yaw = 0;
+
+  /**
+   * @brief Euler rotation axes.
+   * @details The axes about which the Euler rotations are done. Default is
+   * the <b>X-Y-Z Extrinsic (Tait-Bryan)</b> Euler angle convention.
+   */
+  std::array<int, 3> axes = DEFAULT_EULER_ANGLE_AXES;
+
+  /**
+   * @brief Euler frame rotation convention.
+   * @details The Euler frame rotation convention, either \c Extrinsic (static
+   * frame) or \c Intrinsic (rotating frame).
+   *
+   */
+  EulerRotationType rot_type = DEFAULT_EULER_ANGLE_ROTATION_TYPE;
+
+  /**
+   * @brief Construct an Euler Angles object with default values and
+   * convention.
+   */
+  EulerAngles() = default;
+
+  /**
+   * @brief Construct an Euler Angles object with default convention.
+   *
+   * @param rpy_in array of roll, pitch, and yaw angles
+   */
+  explicit EulerAngles(const std::array<Scalar, 3> &rpy_in)
+      : roll(rpy_in[0]), pitch(rpy_in[1]), yaw(rpy_in[2]) {}
+
+  /**
+   * @brief Construct an Euler Angles object with default convention.
+   *
+   * @param rpy_in vector of roll, pitch, and yaw angles
+   */
+  explicit EulerAngles(const Eigen::Vector3<Scalar> &rpy_in)
+      : roll(rpy_in(0)), pitch(rpy_in(1)), yaw(rpy_in(2)) {}
+
+  /**
+   * @brief Construct an Euler Angles object with default convention.
+   *
+   * @param roll_in roll angle
+   * @param pitch_in pitch angle
+   * @param yaw_in yaw angle
+   */
+  EulerAngles(const Scalar &roll_in,
+              const Scalar &pitch_in,
+              const Scalar &yaw_in)
+      : roll(roll_in), pitch(pitch_in), yaw(yaw_in) {}
+
+  /**
+   * @brief Construct an Euler Angles object.
+   *
+   * @param roll_in roll angle
+   * @param pitch_in pitch angle
+   * @param yaw_in yaw angle
+   * @param axes_in array of rotation axes identifiers
+   * @param rot_type_in rotation type (Extrinsic or Intrinsic)
+   */
+  EulerAngles(const Scalar &roll_in,
+              const Scalar &pitch_in,
+              const Scalar &yaw_in,
+              const std::array<int, 3> &axes_in,
+              const EulerRotationType &rot_type_in)
+      : roll(roll_in), pitch(pitch_in), yaw(yaw_in), axes(axes_in),
+        rot_type(rot_type_in) {}
```

Comment:
Are we currently supporting any other convention? if not we may want to remove this and the axes and rot type.
If we did support other conventions, we would need to properly define the constructor to take the angles in the order they were in the convention, e.g. XZY should not construct as EulerAngles euler(X,Y,Z,{0,2,1},EulerAngle::EXTRINSIC)

Reply:
Just need a function that maps to roll pitch yaw via the "axes_in"

Reply:
Resolved in 297e88ce81566e21f82ba644acfb587193045eed